### PR TITLE
chore: backport tasks

### PR DESCRIPTION
**Pull request by iliapolo**
Introduce projen tasks for running manual backports using the [backport](https://github.com/sqren/backport) tool.

For example, running `BACKPORT_PR_NUMBER=555 yarn backport:21` will backport PR number 555 to branch `k8s-21/main`. Required `GITHUB_TOKEN` env variable to be set.

Also, there are a couple of things currently preventing a smooth backport flow:

- In many cases, the import statements in our files cause a merge conflict. This is because resources are named differently on different branches. This PR changes our imports to use a namespaced import for first party code, to reduce changes to import statements.
- The `docs/*.md` files always create a merge conflict because they contain the kuberenets spec version in the element ids, which correspond to the branch name. We don't strictly need these files in source control, we just need them published in the NPM tarball for the website. This PR adds these files to .gitignore for now until we come up with something to workaround this.

### CODE REVIEW

File: .projenrc.js [hunk 1]
```diff
@@ -109,27 +109,23 @@ docgenTask.exec('jsii-docgen -l typescript -o docs/typescript.md');
...
-backport.addJob('backport', {
+const backportWorkflow = project.github.addWorkflow('backport');
+backportWorkflow.on({ pullRequest: { types: ['closed'] } });
+backportWorkflow.addJob('backport', {
   runsOn: 'ubuntu-18.04',
   permissions: {
     contents: JobPermission.WRITE,
   },
   steps: [
-    {
```
Comment (iliapolo) [inline .projenrc.js:125]: We can remove this now that https://github.com/sqren/backport/pull/336 is merged and released

Comment (Chriscbr) [reply .projenrc.js:125]: open source w00t!

**Review by Chriscbr — APPROVED** [review]
> We don't strictly need these files in source control, we just need them published in the NPM tarball for the website.

Ah neat - that's convenient. I think at one point when I ran into these conflicts, I considered some kind of solution that looks at the list of merge conflicts and simply ignores them and regenerates them if the conflicts were with docs files. But your solution is 10x more practical. :)

> For example, running `BACKPORT_PR_NUMBER=555 yarn backport:21` will backport PR number 555 to branch `k8s-21/main`.

nit: `backport-to: 21` or `backport-to-branch:21` instead?

Looks good to me otherwise!

**Comment by iliapolo** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|k8s-21/main|[](https://github.com/cdk8s-team/cdk8s-plus/pull/644)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Comment by iliapolo** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|k8s-20/main|[](https://github.com/cdk8s-team/cdk8s-plus/pull/645)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**